staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -162,144 +162,174 @@ func ValidateCustomResourceDefinitionNames(names *apiextensions.CustomResourceDe
 	return allErrs
 }
 
+// additionalSpecValidator applies validations for different OpenAPI specfication versions.
+type additionalSpecValidator interface {
+	validateCustomResourceDefinitionOpenAPISpec(spec *apiextensions.JSONSchemaProps, fldPath *field.Path) field.ErrorList
+}
+
 // ValidateCustomResourceDefinitionValidation statically validates
-func ValidateCustomResourceDefinitionValidation(CustomResourceValidation *apiextensions.CustomResourceValidation, fldPath *field.Path) field.ErrorList {
+func ValidateCustomResourceDefinitionValidation(customResourceValidation *apiextensions.CustomResourceValidation, fldPath *field.Path) field.ErrorList {
```

Comment:
Yeah, the name looks a bit funny. But `ValidateCustomResourceDefinitionSpec` exists already :/

(https://github.com/kubernetes/kubernetes/pull/47263/files#diff-54fdd67617b0f2ef1b5a09148f77303eR64-R107).